### DISCUSSION TITLE
Phonon dispersion with anharmonic contributions

### DISCUSSION THREAD

**Discussion by RicardoCecconello** [ttadano/alamode#141] [Q&A] [answered]
Dear Dr. Tadano,

I want to calculate and plot the phonon dispersion of a structure taking into account the anharmonic contributions. For that, I followed all the steps until step 5 of the Si tutorial (https://alamode.readthedocs.io/en/latest/tutorial.html). After that, I just want to plot the phonon dispersion. But I'm a little confused. The file si_alm2.in, with MODE = optimize, has the following structure:

&optimize
  DFSET = DFSET_cubic
  FC2XML = si222.xml # Fix harmonic IFCs
/

The FC2XML file, as I understood, is the one calculated in the harmonic approximation. If I want to get the phonon dispersion (considering harmonic and anharmonic contributions, i.e., "all" contributions at the same time), should I still use FC2XML? In doing so, I got the same dispersion (even the .bands files are identical).
Otherwise, if I don't use FC2XML, the dispersion is different. In this case, only the calculations made in the anharmonic approximation are used, right? Is this the right way to do what I'm intending to do?

Thanks in advance,
Ricardo.

**Comment by ttadano** [comment, marked as answer]
By using the `FC2XML` option in **ALM**, you can fix the harmonic force constants when fitting the force constant model to the given training data (`DFSET`). More specifically, when the force constant model is composed of the harmonic and cubic force constants, only the cubic force constants are optimized by fitting, while the harmonic force constants are frozen.

When computing phonons using **ANPHON** with `MODE = phonons`, the code only uses the harmonic force constants in the `FCSXML` file. So, the resulting harmonic phonon dispersion does not change.

If you want to include anharmonic corrections to the phonon dispersion curves, you can try one of the following:

- Run self-consistent phonon calculation `MODE = SCPH`. Requires harmonic and fourth-order force constants as inputs.
- Use temperature-dependent effective potential (TDEP) approach. To use it, you will need to run ab initio molecular dynamics at a target temperature, prepare the training data set (`DFSET`) from the MD trajectory, and fit the harmonic force constant model to the training data. This will give you effective harmonic force constants and effective harmonic phonon frequencies.

**Reply by RicardoCecconello** [reply]
Thank you for the answers!

We'll try the MODE = SCPH approach. In this case, do we still have to calculate the third-order force constants?

Besides this, I've been looking at the PbTe example and the scph.in file only informs one .xml file ( FCSXML = PbTe_anharm444.xml). This one, as I understood, is generated with ALM and MODE = optimize, and DFSET generated only for the fourth-order SCF calculations. Is this right?

And at what step do harmonic force constants have to be given?

**Reply by ttadano** [reply]
For `MODE = SCPH`, you only need harmonic and fourth-order force constants. 

For computing fourth-order force constants, we recommend using displacement-force data generated by AIMD or random displacement methods. If you use this approach, both third-order and fourth-order terms can be fitted simultaneously, and it is more efficient than systematically displacing a few atoms in the supercell (as suggested by `MODE = suggest`).

Sorry for the SCPH tutorial being not very detailed. For more detailed tutorial on SCPH, please see materials at https://github.com/ttadano/alamode_tutorial/tree/main/hands-on_TSUBAME2022.

**Reply by RicardoCecconello** [reply]
Thank you again!

This tutorial will help a lot!
